src/rhsm/config.py

```diff
@@ -120,8 +121,10 @@ def __init__(self, config_file=None, defaults=None):
 
     def save(self, config_file=None):
         """Writes config file to storage."""
-        fo = open(self.config_file, "w")
+        fo = tempfile.NamedTemporaryFile(mode="w", dir="/etc/rhsm", delete=False)
```

Comment:
> Please use dir=os.path.dirname(self.config_file).

Ouch, but of course!

Reply:
> Please use try: except: here, because rhsm is module that can be used by other application. When the application is not executed by root, then it will raise OSError exception.

What should be done the `except:` branch? The current code on master does not use try/except, so maybe we fix that in a separate PR?

Reply:
>> Please use dir=os.path.dirname(self.config_file).
>
> Ouch, but of course!

Done.

Reply:
> > Please use try: except: here, because rhsm is module that can be used by other application. When the application is not executed by root, then it will raise OSError exception.
> 
> What should be done in the `except:` branch? The current code on master does not use try/except, so maybe we fix that in a separate PR?

Good point. rhsm is module and it is not possible to e.g. log anything to rhsm.log. Let's do not use try-except and application using this module will have to handle exceptions.